compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -236,9 +238,15 @@ impl Inliner<'tcx> {
             return false;
         }
 
-        // Avoid inlining functions marked as no_sanitize if sanitizer is enabled,
-        // since instrumentation might be enabled and performed on the caller.
-        if self.tcx.sess.opts.debugging_opts.sanitizer.intersects(codegen_fn_attrs.no_sanitize) {
+        let self_features = &self.codegen_fn_attrs.target_features;
+        let callee_features = &codegen_fn_attrs.target_features;
+        if !callee_features.iter().all(|feature| self_features.contains(feature)) {
```

Comment:
```suggestion
        if callee_features.iter().any(|feature| !self_features.contains(feature)) {
```
Nit: I find these kinds of conditionals easier to parse when they're in normal form (negation on the inside).